src/widgets/PrescriptionExtra.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { DropDown } from './DropDown';
+import { PageInfo } from './PageInfo';
+import { TextEditor } from './modalChildren';
+import { ModalContainer } from './modals/ModalContainer';
+
+import { PrescriptionActions } from '../actions/PrescriptionActions';
+import { pageInfoStrings, modalStrings } from '../localization';
+import {
+  selectTransactionComment,
+  selectPatientType,
+  selectTransactionCategoryName,
+  selectPrescriptionCategories,
+} from '../selectors/prescription';
+import { selectUsingPrescriptionCategories, selectUsingPatientTypes } from '../selectors/modules';
+
+const PATIENT_TYPES = ['Inpatient', 'Outpatient'];
+
+const PrescriptionExtraComponent = ({
+  onOpenCommentModal,
+  onEditComment,
+  onCloseCommentModal,
+  onUpdatePatientType,
+  onUpdateCategory,
+  commentModalOpen,
+  comment,
+  categoryName,
+  patientType,
+  prescriptionCategories,
+  usingPatientTypes,
+  usingPrescriptionCategories,
+}) => {
+  const onCategorySelection = React.useCallback(
+    (_, index) => onUpdateCategory(prescriptionCategories[index]),
+    []
+  );
+
+  const pageInfoColumns = React.useMemo(
+    () => [
+      [
+        {
+          title: `${pageInfoStrings.comment}:`,
+          info: comment,
+          onPress: onOpenCommentModal,
+          editableType: 'text',
+        },
+      ],
+    ],
+    [comment, onOpenCommentModal]
+  );
+
+  return (
+    <View style={localStyles.container}>
+      {usingPatientTypes && (
+        <DropDown
+          values={PATIENT_TYPES}
+          selectedValue={patientType}
+          onValueChange={onUpdatePatientType}
+        />
+      )}
+      {usingPrescriptionCategories && (
+        <DropDown
+          values={prescriptionCategories}
+          selectedValue={categoryName}
+          onValueChange={onCategorySelection}
+        />
+      )}
+      <PageInfo columns={pageInfoColumns} />
+      <ModalContainer
+        fullScreen
+        onClose={onCloseCommentModal}
+        isVisible={commentModalOpen}
+        title={modalStrings.edit_the_prescription_comment}
+      >
+        <TextEditor onEndEditing={x => onEditComment(x)} text={comment} />
```

Comment:
Can this just be `onEndEditing={onEditComment}`?

Reply:
1142a900f96f9b201f5068ec2d2589be2be09ff8 oops